src/core/thread.d

```diff
@@ -2293,7 +2294,7 @@ extern (C) void thread_joinAll()
             Thread.remove(t);
             t = tn;
         }
-        else if (t.isDaemon)
```

Comment:
I think daemon should become an argument to thread start and should not be changed afterwards. But that would be quite some change.